### ISSUE TITLE
Salesforce lightning is not supporting in cypress

### ISSUE THREAD

**Issue by satish151994** [cypress-io/cypress#18935]
### Current behavior

I tried some of the workarounds to open salesforce lightning in cypress, but not success. Could anyone please help or guide is there any configuration setting is needed in salesforce or how i can login into salesforce lightning using cypress script.

### Test code to reproduce
```js
Cypress.Commands.add('login', () => {
    const loginUrl = Cypress.env("salesforce").loginUrl
    const username = Cypress.env("salesforce").username
    const password = Cypress.env("salesforce").password
    cy.request('https://test.salesforce.com/?un=satish.mundiganal@gmail.com.sit&pw="---"&startURL=%2F001')
    .then(() => {
    cy.visit('https://test.salesforce.com/?un=satish.mundiganal@gmail.com.sit&pw=""')
    })
    })
```

### Cypress Version

Cypress package version: 7.4.0 Cypress binary version: 7.4.0 Electron version: 12.0.0-beta.14 Bundled Node version: 14.15.1

**Comment by bahunov** [comment]
@satish151994 - is the login connected to some microsoft login? (AAD) ?

If not then there are several options:
- in some cases you need to save down some localStorage data after cy.request e.g:
```
cy.request({
      url: 'your-end-point',
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: {
      login:"user@name",
      key:"password",
    },
    }).then((resp) => {
      window.localStorage.setItem('some-auth-tokens', `{"authToken":"${resp.body.data.key}"}`);
    });
```
    
    - second option is to login simply through the UI

**Comment by tit** [comment]
@satish151994 use next code for login

```
cy.request({
      body: {
        display: 'page',
        hasRememberUn: true,
        height: '1080',
        local: '',
        locale: '',
        login: '',
        loginType: '',
        loginURL: '',
        lt: 'standard',
        oauth_callback: '',
        oauth_token: '',
        pw: Cypress.env('password'),
        qs: '',
        serverid: '',
        startURL: '',
        un: Cypress.env('email'),
        useSecure: true,
        username: Cypress.env('email'),
        width: '2560',
      },
      form: true,
      method: 'POST',
      url: 'https://login.salesforce.com',
    })

    cy.visit('https://org-name.lightning.force.com/')
```

**Comment by lucascruz9494** [comment]
> @satish151994 use next code for login
> 
> ```
> cy.request({
>       body: {
>         display: 'page',
>         hasRememberUn: true,
>         height: '1080',
>         local: '',
>         locale: '',
>         login: '',
>         loginType: '',
>         loginURL: '',
>         lt: 'standard',
>         oauth_callback: '',
>         oauth_token: '',
>         pw: Cypress.env('password'),
>         qs: '',
>         serverid: '',
>         startURL: '',
>         un: Cypress.env('email'),
>         useSecure: true,
>         username: Cypress.env('email'),
>         width: '2560',
>       },
>       form: true,
>       method: 'POST',
>       url: 'https://login.salesforce.com',
>     })
> 
>     cy.visit('https://org-name.lightning.force.com/')
> ```

could you explain how this works? why there are parameters in blank?  why do uou use this url https://login.salesforce.com ?

**Comment by AtofStryker** [comment]
@satish151994 can you check out https://github.com/cypress-io/cypress/issues/24290#issuecomment-1376214221 and see if it might solve your issue? You should be able to log in with the API/SOAP programmatically or with `cy.origin()` on the login page. I also pushed up some sample tests [here](https://github.com/AtofStryker/cypress-issue-23958/tree/sample-tests-salesforce/cypress)

**Comment by manuelfernandez90** [comment]
@satish151994 To login through the API you can do it by creating a custom command in "commands.js" where you first execute an sfdx command line to get the instanceUrl and the accessToken, and then call the API with these values. You can only do this with an "admin" user and you must have previously installed sfdx-cli (https://www.npmjs.com/package/sfdx-cli).

![image](https://user-images.githubusercontent.com/104433705/213664517-551a6d86-b4b3-49a2-9572-2b94e749479b.png)

Then you can use cy.login() in your test

**Comment by AtofStryker** [comment]
I believe this issue is solved with the release of `12.4.0` with the support of [experimentalSkipDomainInjection](/references/experiments#Experimental-Skip-Domain-Injection). I am going to close this issue as resolved, but if there are still issues, please let us know!

**Comment by Braholka** [comment]
Ainda não consigo fazer rodar

**Comment by AtofStryker** [comment]
> Ainda não consigo fazer rodar (I still can't get it to run)

@Braholka would you be able to share some type of reproduction repository so we can diagnose the issue to see what is happening?

**Comment by thsrv** [comment]
> I believe this issue is solved with the release of `12.4.0` with the support of [experimentalSkipDomainInjection](/references/experiments#Experimental-Skip-Domain-Injection). I am going to close this issue as resolved, but if there are still issues, please let us know!

@AtofStryker Include it in the cypress config, but I can log in but the test does not advance to the next step. The execution is stuck on a "new url" this problem only happens in Chrome, in Electron it runs normally. It is a requirement in my organization that the tests be in Chrome.

![image](https://github.com/user-attachments/assets/07b30d74-00fc-4f67-b85c-b2d43da1157b)

**Comment by thsrv** [comment]
> > I believe this issue is solved with the release of `12.4.0` with the support of [experimentalSkipDomainInjection](/references/experiments#Experimental-Skip-Domain-Injection). I am going to close this issue as resolved, but if there are still issues, please let us know!
> 
> @AtofStryker Include it in the cypress config, but I can log in but the test does not advance to the next step. The execution is stuck on a "new url" this problem only happens in Chrome, in Electron it runs normally. It is a requirement in my organization that the tests be in Chrome.
> 
> ![image](https://private-user-images.githubusercontent.com/43561909/349642927-07b30d74-00fc-4f67-b85c-b2d43da1157b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNDUxMTYsIm5iZiI6MTcyMTI0NDgxNiwicGF0aCI6Ii80MzU2MTkwOS8zNDk2NDI5MjctMDdiMzBkNzQtMDBmYy00ZjY3LWI4NWMtYjJkNDNkYTExNTdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDE5MzMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MGJiYTUxMzEyODAxMjY3ZjhkMjA1YWJmOWEzNjNlOTE0NTkwNjhiMzc2MzgyODU3MzhkM2JjZjE5NTYyZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x9GoSsIhDYorjxzvkcCJ9CWFDgPywZ0cf4HPNbFYKr4)

You can see that it is not redirecting

**Comment by AtofStryker** [comment]
> > > I believe this issue is solved with the release of `12.4.0` with the support of [experimentalSkipDomainInjection](/references/experiments#Experimental-Skip-Domain-Injection). I am going to close this issue as resolved, but if there are still issues, please let us know!
> > 
> > 
> > @AtofStryker Include it in the cypress config, but I can log in but the test does not advance to the next step. The execution is stuck on a "new url" this problem only happens in Chrome, in Electron it runs normally. It is a requirement in my organization that the tests be in Chrome.
> > ![image](https://private-user-images.githubusercontent.com/43561909/349642927-07b30d74-00fc-4f67-b85c-b2d43da1157b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNDUxMTYsIm5iZiI6MTcyMTI0NDgxNiwicGF0aCI6Ii80MzU2MTkwOS8zNDk2NDI5MjctMDdiMzBkNzQtMDBmYy00ZjY3LWI4NWMtYjJkNDNkYTExNTdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDE5MzMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MGJiYTUxMzEyODAxMjY3ZjhkMjA1YWJmOWEzNjNlOTE0NTkwNjhiMzc2MzgyODU3MzhkM2JjZjE5NTYyZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x9GoSsIhDYorjxzvkcCJ9CWFDgPywZ0cf4HPNbFYKr4)
> 
> You can see that it is not redirecting

@thsrv I am unable to see that image. Are you able to help me with a reproduction so I can reproduce the issue?

**Comment by thsrv** [comment]
> > > > I believe this issue is solved with the release of `12.4.0` with the support of [experimentalSkipDomainInjection](/references/experiments#Experimental-Skip-Domain-Injection). I am going to close this issue as resolved, but if there are still issues, please let us know!
> > > 
> > > 
> > > @AtofStryker Include it in the cypress config, but I can log in but the test does not advance to the next step. The execution is stuck on a "new url" this problem only happens in Chrome, in Electron it runs normally. It is a requirement in my organization that the tests be in Chrome.
> > > ![image](https://private-user-images.githubusercontent.com/43561909/349642927-07b30d74-00fc-4f67-b85c-b2d43da1157b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEyNDUxMTYsIm5iZiI6MTcyMTI0NDgxNiwicGF0aCI6Ii80MzU2MTkwOS8zNDk2NDI5MjctMDdiMzBkNzQtMDBmYy00ZjY3LWI4NWMtYjJkNDNkYTExNTdiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE3VDE5MzMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1MGJiYTUxMzEyODAxMjY3ZjhkMjA1YWJmOWEzNjNlOTE0NTkwNjhiMzc2MzgyODU3MzhkM2JjZjE5NTYyZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.x9GoSsIhDYorjxzvkcCJ9CWFDgPywZ0cf4HPNbFYKr4)
> > 
> > 
> > You can see that it is not redirecting
> 
> @thsrv I am unable to see that image. Are you able to help me with a reproduction so I can reproduce the issue?

@AtofStryker Sorry, the flow I'm testing is that I access Salesforce in the test sandbox, when I enter the username and password, it logs in successfully, but in the log it shows a new url but the redirection does not happen. Execution gets stuck and does not advance to the next step, the test does not fail and remains in a loop. This only happens with Chrome, in the Electron browser it successfully performs all the steps.

![image][Sem título](https://github.com/user-attachments/assets/78269950-4dad-4183-8ed9-56008635aa9b)

config

    `e2e: {
        baseUrl: process.env.baseUrl,
        experimentalModifyObstructiveThirdPartyCode: true,
        experimentalSkipDomainInjection: ['*.salesforce.com', '*.force.com','*.sandbox.lightning.force.com', '*file.force.com'],
        numTestsKeptInMemory: 0,
        supportFile: "cypress/support/e2e.js",
        experimentalMemoryManagement: true,
        chromeWebSecurity: false,`


method login

`

    accessAndLoginSalesforce(user, password){
        cy.visit('/');
        cy.document().its('readyState').should('equal', 'complete');
        cy.get(loginMap.usernameInput).type(user);
        cy.get(loginMap.passwordInput).type(password,{log:false});
        cy.wait(1000);
        cy.get(loginMap.buttonLogin).should('be.visible').click(); 
    }

`

next step


```
   //is accessing an object in the navigation bar
   acessMenuNavBar(item){
        cy.xpath(`//*[@role='navigation']//a[@title='${item}']`).should('be.visible').click({force: true});
        cy.wait(2000);
    }

```



URL https://xxxx--test.sandbox.my.salesforce.com
new URL of redirection https://ORGXXX--test.sandbox.file.force.com/secur/contentDoor?startURL=https%3A%2F%2ORGXXX--test.sandbox.my.salesforce.com%2.....................

**Comment by jennifer-shehane** [comment]
This experimental flag will be made default behavior in Cypress 14 with `injectDocumentDomain` being `false`. https://github.com/cypress-io/cypress/pull/30770
